packages/fluentui/react-northstar-prototypes/src/prototypes/compactChat/CompactChatSlots.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+
+import { EditIcon, EmojiIcon, EyeFriendlierIcon, LikeIcon, RedbangIcon } from '@fluentui/react-icons-northstar';
+import { Avatar, Chat, Provider, ReactionProps, ShorthandCollection } from '@fluentui/react-northstar';
+
+import { robinAvatar } from './compactAvatars';
+
+const reactions: ShorthandCollection<ReactionProps> = [
+  {
+    key: 'up',
+    icon: <LikeIcon />,
+    content: '1',
+  },
+  {
+    key: 'smile',
+    icon: <EmojiIcon />,
+    content: '5',
+  },
+];
+
+const slotLabelStyles = (label: string, slotStyles?, beforeStyles?) => ({
+  border: '1px solid #000000',
+  padding: '12px',
+  position: 'relative',
+  ...slotStyles,
+  ':before': {
+    background: '#000000',
+    bottom: '-1px',
+    color: '#ffffff',
+    content: `'${label}'`,
+    fontSize: '11px',
+    letterSpacing: '0.1px',
+    lineHeight: '9px',
+    opacity: 0.75,
+    paddingBottom: '2px',
+    position: 'absolute',
+    right: '-1px',
+    ...beforeStyles,
+  },
+});
+
+export const CompactChatSlots = () => (
+  <Provider
+    theme={{
+      componentStyles: {
+        ChatItem: {
+          gutter: slotLabelStyles(
+            'gutter',
+            { backgroundColor: '#ff00ff', padding: 0, position: 'absolute' },
+            { bottom: '-11px' },
+          ),
+        },
+        ChatMessage: {
+          root: slotLabelStyles('chat-message-root', { backgroundColor: '#87cefa' }),
+          author: slotLabelStyles('author', { backgroundColor: '#e0ffff', zIndex: 1 }),
+          badge: ({ props: { compact } }) =>
+            slotLabelStyles(
+              'badge',
+              { overflow: 'visible', ...(!compact && { position: 'absolute' }) },
+              { textAlign: 'center', left: '0' },
+            ),
+          compactBody: slotLabelStyles('body', { backgroundColor: '#778bea' }),
+          content: slotLabelStyles('content', { backgroundColor: '#f08080' }),
+          reactionGroup: slotLabelStyles('reactions', { backgroundColor: '#e89d4f' }),
+          timestamp: slotLabelStyles('timestamp', { backgroundColor: '#bb80f0', opacity: 1 }),
+        },
+        ChatMessageHeader: {
+          root: slotLabelStyles('header', { backgroundColor: '#80d080' }),
+        },
+        ChatMessageDetails: {
+          root: slotLabelStyles('details', { backgroundColor: '#ffffe0' }),
+        },
+        ChatMessageReadStatus: {
+          root: slotLabelStyles(
+            'read',
+            { position: 'absolute', padding: undefined },
+            { bottom: undefined, top: '-11px', width: '22px' },
+          ),
+        },
+      },
+    }}
+  >
+    <Chat
```

Comment:
nit: might be interesting to add a label to each variant, even if it might be obvious to ppl who have been working on this for a while: 
![image](https://user-images.githubusercontent.com/20744592/121914720-d2957080-cd32-11eb-8a3c-24e5473ac554.png)

Reply:
Added headers.